pkg/kubelet/cm/dra/types.go

```diff
@@ -26,9 +26,12 @@ type ActivePodsFunc func() []*v1.Pod
 
 // Manager manages all the DRA resource plugins running on a node.
 type Manager interface {
-	// Allocate prepares and assigns resources to a container in a pod. From
-	// the requested resources, Allocate will communicate with the
-	// DRA resource plugin to prepare resources.
+	// PrepareResources prepares resources for a container in a pod.
+	// It communicates with the DRA resource plugin to prepare resources and
+	// returns container annotations to trigger CDI injection.
+	PrepareResources(pod *v1.Pod, container *v1.Container) ([]kubecontainer.Annotation, error)
+
+	// Allocate method is defined to satisfy HintProvider interface
```

Comment:
It makes sense to call this `PrepareResources()` instead of `GetCDIAnnotations()` to stay symmetrical with the `UnprepareResource()` call as well as keep it CDI agnostic. I'm not sure how I feel about directly returning `[]kubecontainer.Annotation` though. I'll need to look at the rest of the code to see how it fits togther, but this feels like a strange thing to return, especially if we eventually move away from annotations, I'd rather keep it out of the interface.

Reply:
> It makes sense to call this PrepareResources() instead of GetCDIAnnotations()

Isn't it already done? I think I removed GetCDIAnnotations() from the Manager interface in this PR.

Reply:
Yes, i was agreeing with the change for the first part of my comment. It's the return type I'm not sure of.

Reply:
should be fixed in https://github.com/pohly/kubernetes/pull/22